superset/dashboards/permalink/api.py

```diff
@@ -0,0 +1,174 @@
+              content:
+                application/json:
+                  schema:
+                    type: object
+                    properties:
+                      key:
+                        type: string
+                        description: The key to retrieve the permanent link data.
+            400:
+              $ref: '#/components/responses/400'
+            401:
+              $ref: '#/components/responses/401'
+            422:
+              $ref: '#/components/responses/422'
+            500:
+              $ref: '#/components/responses/500'
+        """
+        key_type = current_app.config["PERMALINK_KEY_TYPE"]
+        try:
+            item = self.add_model_schema.load(request.json)
+            state = item["state"]
+            key = CreateDashboardPermalinkCommand(
+                actor=g.user, id_or_slug=pk, state=state, key_type=key_type,
+            ).run()
+            http_origin = request.headers.environ["HTTP_ORIGIN"]
+            url = f"{http_origin}/superset/dashboard/p/{key}/"
+            return self.response(201, key=key, url=url)
+        except (ValidationError, DashboardPermalinkInvalidStateError) as ex:
+            return self.response(400, message=str(ex))
+        except (DashboardAccessDeniedError, KeyValueAccessDeniedError,) as ex:
+            return self.response(403, message=str(ex))
+        except DashboardNotFoundError as ex:
+            return self.response(404, message=str(ex))
+
+    @expose("/permalink/<string:key>", methods=["GET"])
+    @protect()
+    @safe
+    @event_logger.log_this_with_context(
+        action=lambda self, *args, **kwargs: f"{self.__class__.__name__}.get",
+        log_to_statsd=False,
+    )
+    def get(self, key: str) -> Response:
+        """Retrives permanent link state for dashboard.
+        ---
+        get:
+          description: >-
+            Retrives dashboard state associated with a permanent link.
+          parameters:
+          - in: path
+            schema:
+              type: string
+            name: key
+          responses:
+            200:
+              description: Returns the stored state.
+              content:
+                application/json:
+                  schema:
+                    type: object
+                    properties:
+                      state:
+                        type: object
+                        description: The stored state
+            400:
+              $ref: '#/components/responses/400'
+            401:
+              $ref: '#/components/responses/401'
+            404:
+              $ref: '#/components/responses/404'
+            422:
+              $ref: '#/components/responses/422'
+            500:
+              $ref: '#/components/responses/500'
+        """
+        try:
+            key_type = current_app.config["PERMALINK_KEY_TYPE"]
+            state = GetDashboardPermalinkCommand(g.user, key, key_type).run()
```

Comment:
Can we use named arguments as you did in `POST`? I love this Python feature by the way.

Reply:
Good idea - done